src/extension/cell.ts

```diff
@@ -319,10 +324,9 @@ export class NotebookCellOutputManager {
       }
 
       this.onFinish = new Promise<void>(async (resolve) => {
-        // todo: see bug note inside refreshTerminal
```

Comment:
Could you please educate me on why we are adding this block of code again?

Reply:
Of course. The code was initially written to make outputs serialization available to refresh a secondary output item (`stdout`) with the state of the notebook terminal's stdout/stderr to be sent to the kernel. However, the strategy broke down when I ran into an issue with the VS Code API `replaceOutput`, which appears to be buggy still and, instead of replacing, does concatenate (wind up with 3 instead of 2 output items). The serializer now backfills the stdout at serialization time instead (which ain't great but works).

At the time, I decided to comment out the code because I knew we could reuse it for the exact problem it's now being used to solve, which is marking the document as "dirty" to signal to the editor it needs saving. I knowingly left this problem unsolved at the time to get an early version shipped.

You'll find another comment block inside the method now explaining the strategy change.